providers/Ototoy/mod.ts

```diff
@@ -0,0 +1,461 @@
+					name: artist.textContent.trim(),
+				});
+			}
+
+			const title = titleSpan.textContent.trim();
+
+			const durationCell = trackRow.querySelectorAll('td')[3];
+			if (!durationCell) continue;
+
+			const duration = durationCell.textContent.trim();
+
+			tracks.push({
+				title: title,
+				artists: artists.length > 0 ? artists : undefined,
+				discNumber: currentDisc,
+				trackNumber: parseInt(trackNumber, 10),
+				duration: parseDuration(duration),
+			});
+		}
+
+		return tracks;
+	}
+
+	// The format is as follows:
+	//
+	// <div class = "album-meta-data">
+	//    <h1 class="album-title">Foo album</h1>
+	//    <p class="album-artist">
+	//       <span class="album-artist">
+	//          <a href="/_/default/a/123456">
+	//          Bar artist
+	//       </span>
+	//    </p>
+	//    <div class="detail">
+	//       <p class="release-day">Release date: 2025-11-24</p>
+	//       <p class="release-day">Original release date: 2025-11-24</p>
+	//       <p class="label-name">Label: <a href="/labels/246810">Baz label</a></p>
+	//       <p class="catalog-id>Catalog number: NKCD-6849</p>
+	//    </div>
+	// </div>
+	//
+	// NOTES:
+	//
+	// * The `label-name` and `catalog-id` are optional
+	// * The release can have an "original" release date, a platform release date, or both. "Release date" is the preferred date.
+	//   * In the case that only one date is present, sometimes "Original" is used, sometimes not. Whatever's available will
+	//     be used.
+	parseAlbumMeta(doc: HTMLDocument): AlbumMeta | undefined {
+		const albumMetadata = doc.querySelector('div.album-meta-data');
+		if (!albumMetadata) return undefined;
+
+		const titleHeading = albumMetadata.querySelector('h1.album-title');
+		if (!titleHeading) return undefined;
+
+		const title = titleHeading.textContent.trim();
+
+		const artistSpans = Array.from(albumMetadata.querySelectorAll('p.album-artist > span.album-artist'));
+		if (artistSpans.length === 0) return undefined;
+
+		const artists: Artist[] = [];
+		for (const span of artistSpans) {
+			const anchor = span.querySelector('a');
+			if (!anchor) return undefined;
+
+			const path = anchor.getAttribute('href');
+			if (!path) return undefined;
+
+			const name = span.textContent.trim();
+
+			artists.push({
+				name,
+				url: `${this.origin}${path}`,
+			});
+		}
+
+		const details = albumMetadata.querySelector('div.detail');
+		if (!details) return undefined;
+
+		let releaseDate: string | undefined;
+		let originalReleaseDate: string | undefined;
+
+		const releaseElements = details.querySelectorAll('p.release-day');
+
+		releaseElements.forEach((el) => {
+			const text = el.textContent.trim();
+			if (text.startsWith('Original release date:')) {
+				originalReleaseDate = text.replace('Original release date:', '').trim();
+			} else if (text.startsWith('Release date:')) {
+				releaseDate = text.replace('Release date:', '').trim();
+			}
+		});
+
+		releaseDate = releaseDate || originalReleaseDate;
```

Comment:
I wonder whether we should have such a fallback, there are already enough providers returning the "wrong" original release date. Hopefully Harmony will be better equipped to handle release dates in the future and flag the accuracy of date values.
In this regard it would be better if the cached format contains both date values to leave the decision up to the conversion logic later.
Additionally the dates should be treated as optional rather than causing an extraction error.

Reply:
AFAICT, every release has a date. It's just that they either have both an original and a platform release date *or* just one.

https://ototoy.jp/_/default/p/1822344 for example only has an original release date, which I assume means that the original release date == platform release date. (The release on MB even has the same date: https://musicbrainz.org/release/fa036443-0131-4c4a-a838-d9d033184859)